### ISSUE TITLE
false positive in `unnecessary_string_interpolations` with nullable strings

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#2658]
From the analyzer code base:

```dart
class _PSNode implements PSNode {
  @override
  final String? text;
  @override
  String toString() => '$text'; // <= LINT (but not safe to remove)
}

```

**Comment by devoncarew** [comment]
+1 - I just hit this when cleaning up code in pkg/vm_service; the lint triggered, but actually removing the interpolation is not safe, as the interpolated variable is of type `String?`, but value is being passed into a `String` type.
